Flaky TestQuerierLabelNamesAndValues

More details [here](https://github.com/grafana/mimir/runs/4023101231?check_suite_focus=true):

```
=== RUN   TestQuerierLabelNamesAndValues/limit_cardinality_response_elements
14:11:09 Starting minio-9000
14:11:09 Starting consul
14:11:10 consul: ==> Starting Consul agent...
14:11:10 consul: Version: '1.8.4'
14:11:10 consul: Node ID: '4c5a8f09-9cd4-7ee5-6803-211fe3dced2e'
14:11:10 consul: Node name: 'consul'
14:11:10 consul: Datacenter: 'dc1' (Segment: '<all>')
14:11:10 consul: Server: true (Bootstrap: false)
14:11:10 consul: Client Addr: [0.0.0.0] (HTTP: 8500, HTTPS: -1, gRPC: 8502, DNS: 8600)
14:11:10 consul: Cluster Addr: 127.0.0.1 (LAN: 8301, WAN: 8302)
14:11:10 consul: Encrypt: Gossip: false, TLS-Outgoing: false, TLS-Incoming: false, Auto-Encrypt-TLS: false
14:11:10 consul: ==> Log data will now stream in as it occurs:
14:11:10 consul: ==> Consul agent running!
14:11:10 Ports for container: e2e-mimir-test-minio-9000 Mapping: map[9000:49592]
14:11:10 Started minio-9000
14:11:10 minio-9000: Attempting encryption of all config, IAM users and policies on MinIO backend
14:11:10 Ports for container: e2e-mimir-test-consul Mapping: map[8500:49594]
14:11:10 Started consul
14:11:13 Starting query-frontend
14:11:14 Ports for container: e2e-mimir-test-query-frontend Mapping: map[80:49596]
14:11:14 Started query-frontend
14:11:14 Starting querier
14:11:14 Starting ingester-1
14:11:14 Starting distributor
14:11:14 Starting ingester-2
14:11:14 Starting ingester-3
14:11:16 Ports for container: e2e-mimir-test-querier Mapping: map[80:49598]
14:11:16 Started querier
14:11:16 ingester-3: level=warn ts=2021-10-27T14:11:16.571179536Z caller=experimental.go:24 msg="experimental feature in use" feature="ingester instance limits"
14:11:16 Ports for container: e2e-mimir-test-ingester-3 Mapping: map[80:49600]
14:11:16 Started ingester-3
14:11:16 ingester-2: level=warn ts=2021-10-27T14:11:16.850069316Z caller=experimental.go:24 msg="experimental feature in use" feature="ingester instance limits"
14:11:17 Ports for container: e2e-mimir-test-distributor Mapping: map[80:49602]
14:11:17 Started distributor
14:11:17 Ports for container: e2e-mimir-test-ingester-2 Mapping: map[80:49604]
14:11:17 Started ingester-2
14:11:17 ingester-1: level=warn ts=2021-10-27T14:11:17.155273574Z caller=experimental.go:24 msg="experimental feature in use" feature="ingester instance limits"
14:11:17 Ports for container: e2e-mimir-test-ingester-1 Mapping: map[80:49606]
14:11:17 Started ingester-1
    querier_label_name_values_test.go:178: 
        	Error Trace:	querier_label_name_values_test.go:178
        	Error:      	Received unexpected error:
        	            	metric not found
        	            	github.com/grafana/mimir/integration/e2e.init
        	            		/home/runner/work/mimir/mimir/integration/e2e/service.go:29
        	            	runtime.doInit
        	            		/opt/hostedtoolcache/go/1.15.15/x64/src/runtime/proc.go:5652
        	            	runtime.doInit
        	            		/opt/hostedtoolcache/go/1.15.15/x64/src/runtime/proc.go:5647
        	            	runtime.doInit
        	            		/opt/hostedtoolcache/go/1.15.15/x64/src/runtime/proc.go:5647
        	            	runtime.doInit
        	            		/opt/hostedtoolcache/go/1.15.15/x64/src/runtime/proc.go:5647
        	            	runtime.main
        	            		/opt/hostedtoolcache/go/1.15.15/x64/src/runtime/proc.go:191
        	            	runtime.goexit
        	            		/opt/hostedtoolcache/go/1.15.15/x64/src/runtime/asm_amd64.s:1374
        	            	metric=cortex_request_duration_seconds service=ingester-1
        	            	github.com/grafana/mimir/integration/e2e.(*HTTPService).SumMetrics
        	            		/home/runner/work/mimir/mimir/integration/e2e/service.go:631
        	            	github.com/grafana/mimir/integration/e2e.(*HTTPService).WaitSumMetricsWithOptions
        	            		/home/runner/work/mimir/mimir/integration/e2e/service.go:577
        	            	github.com/grafana/mimir/integration.TestQuerierLabelNamesAndValues.func1
        	            		/home/runner/work/mimir/mimir/integration/querier_label_name_values_test.go:178
        	            	testing.tRunner
14:11:49 Killing query-frontend
14:11:49 Killing ingester-2
14:11:49 Killing querier
14:11:49 Killing ingester-1
14:11:50 querier: level=error ts=2021-10-27T14:11:50.414907464Z caller=frontend_processor.go:75 msg="error processing requests" address=172.31.0.4:9095 err="rpc error: code = Unavailable desc = error reading from server: EOF"
14:11:50 querier: level=error ts=2021-10-27T14:11:50.501573552Z caller=frontend_processor.go:69 msg="error contacting frontend" address=172.31.0.4:9095 err="rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 172.31.0.4:9095: connect: connection refused\""
14:11:50 querier: level=error ts=2021-10-27T14:11:50.685999883Z caller=frontend_processor.go:69 msg="error contacting frontend" address=172.31.0.4:9095 err="rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 172.31.0.4:9095: connect: connection refused\""
14:11:51 Killed ingester-1
14:11:51 Killed ingester-3
14:11:51 Killed minio-9000
14:11:51 Killed query-frontend
14:11:51 Killed consul
14:11:51 Killed ingester-2
14:11:51 Killed distributor
14:11:51 Killed querier
--- FAIL: TestQuerierLabelNamesAndValues (87.36s)
    --- PASS: TestQuerierLabelNamesAndValues/obtain_label_names_and_values_with_default_selector_and_limit (22.25s)
    --- PASS: TestQuerierLabelNamesAndValues/apply_request_selector (21.86s)
    --- FAIL: TestQuerierLabelNamesAndValues/limit_cardinality_response_elements (21.52s)
    --- FAIL: TestQuerierLabelNamesAndValues/apply_request_selector_and_limit (21.73s)
```

--- a/integration/querier_label_name_values_test.go
+++ b/integration/querier_label_name_values_test.go
@@ -27,8 +27,6 @@ var cardinalityEnvLabelValues = []string{"prod", "staging", "dev"}
 var cardinalityJobLabelValues = []string{"distributor", "ingester", "store-gateway", "querier", "compactor"}
 
 func TestQuerierLabelNamesAndValues(t *testing.T) {
-	t.Skip("Flaky, see https://github.com/grafana/mimir/issues/426")
-
 	const numSeriesToPush = 1000
 
 	// Define response types.
@@ -173,16 +171,6 @@ func TestQuerierLabelNamesAndValues(t *testing.T) {
 			res, err := client.LabelNamesAndValues(tc.selector, tc.limit)
 			require.NoError(t, err)
 			require.Equal(t, http.StatusOK, res.StatusCode)
-
-			// Ensure all ingesters have been invoked.
-			ingesters := []*e2emimir.MimirService{ingester1, ingester2, ingester3}
-			for _, ing := range ingesters {
-				require.NoError(t, ing.WaitSumMetricsWithOptions(
-					e2e.Equals(1),
-					[]string{"cortex_request_duration_seconds"},
-					e2e.WithMetricCount,
-					e2e.WithLabelMatchers(labels.MustNewMatcher(labels.MatchEqual, "route", "/cortex.Ingester/LabelNamesAndValues"))))
-			}
 
 			// Test results.
 			var lbNamesAndValuesResp labelNamesAndValuesResponse

